src/npc.cpp

```diff
@@ -3030,14 +3028,14 @@ void npc::die( Creature *nkiller )
             add_msg( _( "A cold shock of guilt washes over you." ) );
             player_character.add_morale( MORALE_KILLER_HAS_KILLED, -15, 0, 1_days, 1_hours );
         }
-        if( !badguy ) {
+        if( !was_enemy ) {
```

Comment:
This boolean no longer serves a purpose since it is used in one place and it evaluates a single return. It being placed away from the place where it is evaluated(~70 lines later) makes the code harder to read and is generally, just bad practice.